client/keys/migrate_test.go

```diff
@@ -2,41 +2,97 @@ package keys
 
 import (
 	"context"
-	"fmt"
+     "fmt"
 	"testing"
 
+	design99keyring "github.com/99designs/keyring"
+	"github.com/stretchr/testify/suite"
+
 	"github.com/cosmos/cosmos-sdk/client"
+    "github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+   "github.com/cosmos/cosmos-sdk/testutil"
+	"github.com/cosmos/cosmos-sdk/codec"
+)
 
-	"github.com/stretchr/testify/require"
+type setter interface {
+	SetItem(item design99keyring.Item) error
+}
 
-	"github.com/otiai10/copy"
-	"github.com/stretchr/testify/assert"
+type MigrateTestSuite struct {
+	suite.Suite
 
-	"github.com/cosmos/cosmos-sdk/client/flags"
-	"github.com/cosmos/cosmos-sdk/crypto/keyring"
-	"github.com/cosmos/cosmos-sdk/testutil"
-)
+	dir  string
+	n1   string
+	cdc codec.Codec
+	priv cryptotypes.PrivKey
+	pub  cryptotypes.PubKey
 
-func Test_runMigrateCmd(t *testing.T) {
-	kbHome := t.TempDir()
-	clientCtx := client.Context{}.WithKeyringDir(kbHome)
-	ctx := context.WithValue(context.Background(), client.ClientContextKey, &clientCtx)
+}
+
+func (s *MigrateTestSuite) SetupSuite() {
+	s.dir = s.T().TempDir()
+	s.cdc = simapp.MakeTestEncodingConfig().Codec
+	s.n1 = "cosmos"
```

Comment:
`n1` without local context it's too cryptic (1-2 letter names are good for well known abbreviation, or in a local context, like `i := row.Index`, but not otherwise).